File: onyx-database-tests/test/embedded/encryption/EncryptionTest.java
package embedded.encryption;

import category.EmbeddedDatabaseTests;
import com.onyx.interactors.encryption.DefaultEncryptionInteractor;
import org.junit.Assert;
import org.junit.Test;
import org.junit.experimental.categories.Category;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.UnsupportedEncodingException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.security.GeneralSecurityException;

/**
 *
 * @author cosbor11
 */
@Category({ EmbeddedDatabaseTests.class })
public class EncryptionTest {
    
    protected static final String FILE_LOCATION = "C:/Sandbox/Onyx/Tests/encryption";

    @Test
    public void shouldEncrypt() throws GeneralSecurityException, UnsupportedEncodingException {
        String encryptedText = DefaultEncryptionInteractor.INSTANCE.encrypt("adminadmin");
        Assert.assertTrue(encryptedText != null);
    }

    @Test
    public void shouldDycrypt() throws GeneralSecurityException, IOException {
        String str = "adminadmin";
        String encryptedText = DefaultEncryptionInteractor.INSTANCE.encrypt(str);
        String decryptedText = DefaultEncryptionInteractor.INSTANCE.decrypt(encryptedText);
        
        Assert.assertTrue(str.equals(decryptedText));
       
    }
    
    @Test
    public void shouldFail() throws GeneralSecurityException, IOException {
        String str = "adminadmin";
        String str2 = "adminpassword"; 
        String decryptedText = DefaultEncryptionInteractor.INSTANCE.decrypt(DefaultEncryptionInteractor.INSTANCE.encrypt(str2));
        
        Assert.assertFalse(str.equals(decryptedText));   
    }
    
    
    @Test
    public void shouldWriteToFile() throws GeneralSecurityException, IOException {
       String textToSave = "asdfasdf";
       
       //Create a temporary file
       final File file = new File(FILE_LOCATION + "/tmp");
       file.getParentFile().mkdirs();
       file.createNewFile();
       //Next, write the encrypted bytes to the file.
       
        try (FileOutputStream fileStream = new FileOutputStream(file)) {
            fileStream.write(DefaultEncryptionInteractor.INSTANCE.encrypt(textToSave).getBytes(StandardCharsets.UTF_8));
            fileStream.close();
        }

        String savedText = new String(Files.readAllBytes(Paths.get(file.getAbsolutePath())));

        //make assertions
        Assert.assertEquals(savedText, DefaultEncryptionInteractor.INSTANCE.encrypt(textToSave));
        
        //cleanup file
        boolean delete = file.delete();
        //Now lets read from the file
        
    }
  

}

File: onyx-database-tests/test/encryption/EncryptionTest.kt
package encryption

import com.onyx.interactors.encryption.impl.DefaultEncryptionInteractorInstance
import org.junit.Test

import java.io.File
import java.io.FileOutputStream
import java.nio.charset.StandardCharsets
import java.nio.file.Files
import java.nio.file.Paths
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals
import kotlin.test.assertNotNull

/**
 *
 * @author Chris Osborn
 */
class EncryptionTest {

    @Test
    fun shouldEncrypt() {
        val encryptedText = DefaultEncryptionInteractorInstance.encrypt("admin_admin")
        assertNotNull(encryptedText, "Encrypted text should not be null")
    }

    @Test
    fun shouldDecrypt() {
        val str = "admin_admin"
        val encryptedText = DefaultEncryptionInteractorInstance.encrypt(str)
        val decryptedText = DefaultEncryptionInteractorInstance.decrypt(encryptedText!!)
        assertEquals(str,  decryptedText, "Decrypted text should match original")
    }

    @Test
    fun shouldFail() {
        val str = "admin_admin"
        val str2 = "admin_password"
        val decryptedText = DefaultEncryptionInteractorInstance.decrypt(DefaultEncryptionInteractorInstance.encrypt(str2)!!)

        assertNotEquals(str, decryptedText, "Decrypted should not match")
    }

    @Test
    fun shouldWriteToFile() {
        val textToSave = "asdfasdf"

        //Create a temporary file
        val file = File(FILE_LOCATION + "/tmp")
        file.parentFile.mkdirs()
        file.createNewFile()
        //Next, write the encrypted bytes to the file.

        FileOutputStream(file).use { fileStream ->
            fileStream.write(DefaultEncryptionInteractorInstance.encrypt(textToSave)!!.toByteArray(StandardCharsets.UTF_8))
            fileStream.close()
        }

        val savedText = String(Files.readAllBytes(Paths.get(file.absolutePath)))

        //make assertions
        assertEquals(savedText, DefaultEncryptionInteractorInstance.encrypt(textToSave), "Encrypted text does not match")

        //cleanup file
        file.delete()
    }

    companion object {
        private val FILE_LOCATION = "C:/Sandbox/Onyx/Tests/encryption"
    }
}
